Cookbook/Proposals/SimpleSideEffectInReactiveFeedback.md

```diff
@@ -0,0 +1,124 @@
+
+This is a companion proposal of [Mealy-Machine ReactiveFeedback](https://github.com/Babylonpartners/ios-playbook/pull/98).
+
+## Motivation
+
+Managing side-effect is the most important and unpredictably difficult part in iOS app development.
+Fortunately, we are currently using a one-stop [ReactiveFeedback](https://github.com/Babylonpartners/ReactiveFeedback) to control states, side-effects, and feedback loops to define a whole system (or subsystem per view controller) behavior.
+However, we still have a problem about **how to handle (simple) side-effects that don't require to send a next feedback event**.
+For example, **logging** and **analytics event** are part of this discussion.
+
+A typical implementation of analytics event goes like this:
+
+```swift
+class ViewModel {
+    let (userAction, input) = Signal.pipe()
+    let state: Property<State>
+    let routes: Signal<Route>
+
+    init(analytics: Analytics, ...) {
+        analytics.track(TrackingEvent.initialized)
+
+        /* ReactiveFeedback setup */
+        self.state = Property(
+            ...,
+            feedbacks: [
+                whenButtonTapped(analytics: analytics)
+            ]
+        )
+
+        self.routes = state.signal
+            .filterMap { $0.route }
+            .on(value: { route in
+                analytics.track(TrackingEvent.routingByFeedback(route))
+            })
+
+        userAction
+            .filterMap(Route.init(userAction:))
+            .observeValues { route in
+                analytics.track(TrackingEvent.routingByUser(route))
+            }
+    }
+
+    func send(action: UserAction) {
+        input.observer(Event.ui(action))
+        analytics.track(TrackingEvent.userAction(action)
+    }
+}
+
+// MARK: - Feedbacks
+
+func whenButtonTapped(analytics: Analytics) -> Feedback {
+    return Feedback { signal in
+        return signal
+            .flatMap(.latest) { someAPI }
+            .on(value: { response in
+                analytics.track(TrackingEvent.apiComplete(response))
+            })
+            .map(Event.didFinishAPI)
+    }
+}
+```
+
+Please notice **how `analytics.track()`s are scattered across `ViewModel`**, mostly outside of `RAF`'s feedback system.
+This is because FRP e.g. `ReactiveSwift` is **too easy to inject side-effects anywhere** by using `on(value:)` and `observeValues`.
+While maintaining these side-effects inside `ViewModel` (rather than `RAF`) could be sufficient, "where to implement side-effects" ambiguity still remains, which is not ideal for practicing `RAF` architecture.
+
+In contrast, other architectures e.g. [Elm](https://elm-lang.org) and [Redux](https://redux.js.org) focuses their side-effect handling in `Feedback` layer (`Cmd` and `Middleware`) that _may_ send next feedback event.
+It's worth noting that **non-feedback type `Feedback<Never>` is a subset of `Feedback<Event>`**, thus thinking `Feedback` type as a side-effect handler is more important than using it as a feedback loop.
```

Comment:
Agree. It is interesting how RAF favours some side effects, like making a network call that will result in the state change (or not, as the state can stay the same!) and completely ignores other side effects, which may as well make a network call, but will not result in the state change (or to be more correct, will not be connected with the state input of the system like other feedbacks are and instead will just "go into the ground")